panc/src/main/scripts/panlint/panlint.py

```diff
@@ -99,9 +99,16 @@ def whitespace_around_operators(self, line, string_ranges):
 
             valid = True
             if op == '-' and \
-               re.search(r'\W\s*$', chars_before) and \
-               re.search(r'^\s*\d+\s*\W', chars_after):
+               ((re.search(r'[^\w\s)=]\s*$', chars_before) and re.search(r'^\s*\d+\s*[^\w\s]', chars_after)) or \
+                (re.search(r'=\s*$', chars_before) and re.search(r'^\s*\d+', chars_after))):
```

Comment:
Line joining is implicit within brackets, so more readable (and fewer code-climate errors) would be:
```python
if op == '-' and (
        (re.search(r'[^\w\s)=]\s*$', chars_before) and re.search(r'^\s*\d+\s*[^\w\s]', chars_after))
        or
        (re.search(r'=\s*$', chars_before) and re.search(r'^\s*\d+', chars_after))
    ):
    … code continues here …
```

Or you even go as far as:
```python
if op == '-' and (
        (
            re.search(r'[^\w\s)=]\s*$', chars_before)
            and
            re.search(r'^\s*\d+\s*[^\w\s]', chars_after)
        )
        or
        (
            re.search(r'=\s*$', chars_before)
            and
            re.search(r'^\s*\d+', chars_after)
        )
    ):
    … code continues here …
```
But maybe that's going a bit too far.